### ISSUE TITLE
Flaky tests possibly related to authentication or rate limiting

### ISSUE THREAD

**Issue by brassy-endomorph** [scidsg/hushline#587]
**Describe the bug**

As the title says, some tests are flaky and the errors are sufficiently swallowed and not logged so that there is no indication what is actually going wrong.

We should probably refactor the tests to not use requests to generate users but to instead use fixtures. This will speed the tests up significantly too as a nice bonus.

**Comment by brassy-endomorph** [comment]
Example:
https://github.com/scidsg/hushline/actions/runs/10942529787/job/30379765896?pr=588

```
=================================== FAILURES ===================================
_____________________________ test_change_password _____________________________

client = <FlaskClient <Flask 'hushline'>>

    def test_change_password(client: FlaskClient) -> None:
        username = "test_change_password"
        original_password = f"{token_urlsafe(16)}!"
        new_password = f"{token_urlsafe(16)}!!!"
    
>       user = register_user(client, username, original_password)

tests/test_settings.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

client = <FlaskClient <Flask 'hushline'>>, username = 'test_change_password'
password = 'FznbXfIrnRAUOblBGnKuUw!'

    def register_user(client: FlaskClient, username: str, password: str) -> User:
        # Prepare the environment to not require invite codes
        os.environ["REGISTRATION_CODES_REQUIRED"] = "False"
    
        user_data = {"username": username, "password": password}
    
        response = client.post("/register", data=user_data, follow_redirects=True)
        assert response.status_code == 200
>       assert b"Registration successful!" in response.data
E       AssertionError

tests/auth_helper.py:20: AssertionError
```

**Comment by brassy-endomorph** [comment]
> `password = 'FznbXfIrnRAUOblBGnKuUw!'`

Generated token didn't have a number thus wasn't a valid password. This is the root cause.

**Comment by brassy-endomorph** [comment]
Likely fixed by a0e0a1f. Closing for now.
